### PR TITLE
Unregister command + game prepare scenario

### PR DESCRIPTION
**Pull request by Luke-Bytes**

### CODE REVIEW

File: src/database/GameInstance.ts [hunk 1]
```diff
@@ -175,23 +207,28 @@ export class GameInstance {
...
+    const undecidedPlayers = Array.from(this.getPlayersOfTeam("UNDECIDED"));
+    const shuffled = undecidedPlayers.sort(() => Math.random() - 0.5);
+    const half = Math.ceil(shuffled.length / 2);
+
+    return {
+      BLUE: shuffled.slice(0, half),
+      RED: shuffled.slice(half),
+    };
+  }
+
```
Comment (99-i) [inline src/database/GameInstance.ts:231]: i moved this logic from GameInstance into RandomTeamPickingSession in my branch, i think it's better suited there

File: src/database/prismaClient.ts [hunk 2]
```diff
@@ -78,9 +69,36 @@ export const prismaClient = new PrismaClient().$extends({
...
+          return { error: false };
+        } catch (error) {
+          console.error(
+            `Error removing player ${discordSnowflake} from games:`,
+            error
+          );
+          return {
+            error: "Failed to remove the player from the current game.",
+          };
+        }
```
Comment (99-i) [inline src/database/prismaClient.ts:101]: we don't actually commit to GameParticipation during the game, but only after. i think this is unnecessary. also, a GameParticipation record is meant to be for past games, so we can view match history, it doesn't store anything about the current match, so removing all GameParticipations with a player id will just wipe their game history.

File: src/logic/teams/RandomTeamPickingSession.ts
Comment (99-i) [inline src/logic/teams/RandomTeamPickingSession.ts:1]: i fixed the RandomTeamPickingSession in my draft branch, i moved all the shuffling logic from GameInstance into this class itself, and just set the values in the current GameInstance once they are accepted.

**Review by 99-i — CHANGES_REQUESTED** [review]
changes commented on

File: src/database/prismaClient.ts (hunk 2 above)
Comment (Luke-Bytes) [reply src/database/prismaClient.ts:101]: ah ok lets remove

**Pull request merged**